vendor/k8s.io/kubernetes/pkg/client/restclient/config.go

```diff
@@ -185,6 +189,9 @@ func RESTClientFor(config *Config) (*RESTClient, error) {
 	var httpClient *http.Client
 	if transport != http.DefaultTransport {
 		httpClient = &http.Client{Transport: transport}
+		if config.Timeout != 0 {
```

Comment:
Updated this here: https://github.com/openshift/origin/pull/11104/commits/4a7769ff0233bb104041c3dc8430f14a4fb12c22